scripts/core/editor3/reducers/toolbar.tsx

```diff
@@ -244,4 +248,59 @@ const setPopup = (state: IEditorStore, {type, data}) => {
     return {...state, editorState: newEditorState, popup: {type, data}};
 };
 
+function changeCase(state: IEditorStore, payload: {changeTo: ITextCase, selection: SelectionState}) {
+    const getChangedText = (text: string) => {
+        if (changeTo === 'uppercase') {
+            return text.toUpperCase();
+        } else if (changeTo === 'lowercase') {
+            return text.toLowerCase();
+        } else {
+            assertNever(changeTo);
+        }
+    };
+
+    const {selection, changeTo} = payload;
+    const contentState = state.editorState.getCurrentContent();
+
+    let ended = false;
+    let started = false;
+    const startKey = selection.getStartKey();
+    const endKey = selection.getEndKey();
+
+    const nextBlockMap = state.editorState.getCurrentContent().getBlockMap().map((block) => {
```

Comment:
just wondering - why not using [Modifier.replaceText](https://draftjs.org/docs/api-reference-modifier#replacetext) ?

Reply:
To maintain inline styles I'd need to call `Modifier.replaceText` for every character. On top of that, it requires a `Selection` as one of the arguments, which means I'd need to fiddle with the offsets to account for backward selection.

Reply:
I could use start/end offsets for anchor/focus offset and always create a selection with `backwards: false`, but I'd still need to call it for every char. Replacing ranges seems simpler. Are there any trade-offs you know I'm missing by not using replace-text?

Reply:
not really, seemed simpler, but I was assuming those inline styles could be an issue, so wanted to check